app/models/payment_intent.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+class PaymentIntent < ApplicationRecord
+  belongs_to :holder, polymorphic: true
+  belongs_to :payment_record, polymorphic: true
+  belongs_to :initiated_by, class_name: 'User' # For now only users can initiate payments - in future, this may become a polymorphic association
+  belongs_to :confirmation_source, polymorphic: true, optional: true
+  belongs_to :cancellation_source, polymorphic: true, optional: true
+
+  validate :valid_status_combination
+
+  scope :unpaid, -> { where.not(wca_status: 'succeeded') }
+  scope :started, -> { where.not(wca_status: ['created']) }
+
+  # TODO: Refactor this or move it into this class
```

Comment:
I'm inclined to leave this for another PR - this one already has a LOT of changes